skyve-ext/src/main/java/org/skyve/impl/util/WebStatsUtil.java

```diff
@@ -2,43 +2,155 @@
+			+ "The IP address has changed from %s to %s. "
+			+ "If this change is unexpected, it may indicate unauthorized access to the account. "
+			+ "Please review the login activity and consider updating the user's security settings if necessary.";
+	
+	private static final String COUNTRY_CHANGE_LOG_MESSAGE = 
+			"The user %s has logged in from a different country. "
+			+ "Their location has changed from %s to %s. "
+			+ "If this change is unexpected, it might indicate unauthorized access. Please review the user's recent activity for any discrepancies.";
+	
+	private static final String UNUSUAL_LOGIN_EMAIL_BODY = 
+			"Dear %s,\n\n"
+			+ "We have detected a login attempt to your account from a new location: %s. "
+			+ "If this was you, there's no need to take further action.\n\n"
+			+ "However, if you do not recognize this activity, we strongly recommend that you change your password immediately to secure your account. "
+			+ "You can do this by logging into your account and navigating to <strong>Admin -> Password</strong>.\n\n"
+			+ "For your safety, please do not share your password with anyone.\n\n"
+			+ "If you have any questions or need assistance, please contact our support team at %s.\n\n"
+			+ "Best regards,\n"
+			+ "The Security Team";
+	
+	private static final String UNUSUAL_LOGIN_EMAIL_FAILURE_LOG_MESSAGE = 
+			"Failed to send security alert email to %s at %s regarding a login from a different country. Exception: %s";
+
+
 	private WebStatsUtil() {
 		// do nothing
 	}
 	
-	public static void recordLogin(User user)
-	throws Exception {
+	/**
+	 * This method is used to populate the UserLoginRecord table with details of a new login to the system. It also checks the
+	 * previous record and sends alerts if there's a change in Ip Address and/or country
+	 * 
+	 * @param user
+	 * @param userIPAddress
+	 * @throws Exception
+	 */
+	public static void recordLogin(User user, String userIPAddress)
+			throws Exception {
 		Customer customer = user.getCustomer();
 		Module module = customer.getModule(AppConstants.ADMIN_MODULE_NAME);
-		Document loginRecordDocument = module.getDocument(customer, "UserLoginRecord");
+		Document loginRecordDocument = module.getDocument(customer, AppConstants.USER_LOGIN_RECORD_DOCUMENT_NAME);
 		AbstractPersistentBean loginRecord = loginRecordDocument.newInstance(user);
-		BindUtil.set(loginRecord, "userName", user.getName());
-		BindUtil.set(loginRecord, "loginDateTime", new DateTime(System.currentTimeMillis()));
-		BindUtil.set(loginRecord, "failed", Boolean.FALSE);
+		BindUtil.set(loginRecord, AppConstants.USER_NAME_ATTRIBUTE_NAME, user.getName());
+		BindUtil.set(loginRecord, LOGIN_DATE_TIME, new DateTime(System.currentTimeMillis()));
+		BindUtil.set(loginRecord, FAILED, Boolean.FALSE);
+		BindUtil.set(loginRecord, IP_ADDRESS, userIPAddress);
+
+		// Check if the IpInfo token has been set so as to get the country code
+		String ipInfoToken = UtilImpl.IP_INFO_TOKEN;
```

Comment:
What is the point in storing this in a String? Why can't you just check if `UtilIImpl.IP_INFO_TOKEN` is not null on line 90? You aren't re-using this variable anywhere else that I can see.